### PR TITLE
LOG-682: Remove _output directory, simplify Makefile

### PR DESCRIPTION
**Pull request by alanconway**
Changes:
- trivial source file changes from gofmts and to satisfy the linter.
- drop _output file, not necessary and causes confusion.  
- remove unused variables, simplify targets.  
- add dependencies to make targets self-sufficient.

Signed-off-by: Alan Conway <aconway@redhat.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -64,38 +43,34 @@ run:
...
 
-image: imagebuilder
+image: bin/imagebuilder build
 	@if [ $${USE_IMAGE_STREAM:-false} = false ] && [ $${SKIP_BUILD:-false} = false ] ; \
 	then hack/build-image.sh $(IMAGE_TAG) $(IMAGE_BUILDER) $(IMAGE_BUILDER_OPTS) ; \
 	fi
 
-lint:
+lint: fmt
 	golangci-lint run -c golangci.yaml
```
Comment (igor-karpukhin) [inline Makefile]: @alanconway `make build` will fail if this `golangci-lint` is missing. I guess we should add this as a prerequisite to README.md, or add the way to download it automatically if it is missing.

File: Makefile [hunk 2]
```diff
@@ -1,58 +1,37 @@
...
-IMAGE_TAG?=quay.io/openshift/origin-$(APP_NAME):latest
-export IMAGE_TAG
-MAIN_PKG=cmd/manager/main.go
-export OCP_VERSION?=$(shell basename $(shell find manifests/  -maxdepth 1  -not -name manifests -type d))
+export IMAGE_TAG?=quay.io/openshift/origin-$(APP_NAME):latest
+
+export OCP_VERSION?=4.5
 export CSV_FILE=$(CURDIR)/manifests/$(OCP_VERSION)/cluster-logging.v$(OCP_VERSION).0.clusterserviceversion.yaml
 export NAMESPACE?=openshift-logging
 export EO_CSV_FILE=$(CURDIR)/vendor/github.com/openshift/elasticsearch-operator/manifests/$(OCP_VERSION)/elasticsearch-operator.v$(OCP_VERSION).0.clusterserviceversion.yaml
```
Comment (periklis) [inline Makefile]: IIRC `CSV_FILE` and `EO_CSV_FILE` are unused.

File: Makefile (hunk 1 above)
Comment (alanconway) [reply Makefile]: +1, fixing.

File: Makefile (hunk 2 above)
Comment (alanconway) [reply Makefile]: +1 removed.

File: Dockerfile [hunk 3]
```diff
@@ -12,12 +12,12 @@ RUN INSTALL_PKGS=" \
     yum clean all && \
     mkdir /tmp/ocp-clo && \
     chmod og+w /tmp/ocp-clo
-COPY --from=builder /go/src/github.com/openshift/cluster-logging-operator/_output/bin/cluster-logging-operator /usr/bin/
+COPY cmd/manager /usr/bin/cluster-logging-operator
```
Comment (ewolinetz) [inline Dockerfile]: this defeats the whole purpose of using a builder image

Comment (ewolinetz) [reply Dockerfile]: i think this would break ART -- they do not have access to our `Makefile`

Comment (jcantrill) [reply Dockerfile]: This will break the ART process because this copies the binary from local instead of the building layer.

Comment (alanconway) [reply Dockerfile]: OK, will fix. My bad.

**Comment by alanconway** [comment]
@jcantrill @ewolinetz Fixed my blunder with the Dockerfile, I think this is good now assuming CI passes.

File: Makefile [hunk 4]
```diff
@@ -112,10 +93,10 @@ deploy-elasticsearch-operator:
 deploy-example: deploy
 	oc create -n $(NAMESPACE) -f hack/cr.yaml
 
-test-unit: fmt
-	@LOGGING_SHARE_DIR=$(CURDIR)/files go test $(TEST_OPTIONS) $(PKGS)
+test-unit: generate
+	go test ./pkg/...
 
-test-e2e:
+test-e2e: build
```
Comment (jcantrill) [inline Makefile]: We should rethink the dependency upon `build` as its unnecessary.  In a CI context, the operator will be deployed using the image that was built by a separate CI job.  Locally the deployment will reference an image that only exists if you run `make deploy-image`.  This means the primary place this  is used we are building the operator twice.

File: README.md [hunk 5]
```diff
@@ -46,24 +46,23 @@ Running locally outside an OKD cluster:
...
+See also: [sdk_setup.sh](https://raw.githubusercontent.com/openshift/origin-aggregated-logging/master/hack/sdk_setup.sh) to set up your environment.
 
 The deployment can be optionally modified using any of the following:
 
 | Env Var | Default | Description|
 |---------|---------|------------|
-|`IMAGE_BUILDER`|`imagebuilder`| The command to build the container image|
 |`EXCLUSIONS`|none|The list of manifest files that should will be ignored|
-|`OC`|`oc` in `PATH`| The openshift binary to use to deploy resources|
 |`REMOTE_REGISTRY`|false|`true` if you are running the cluster on a different machine than the one you are developing on|
```
Comment (jcantrill) [inline README.md]: While you are updating, can you remove the ref here and below to "REMOTE_REGISTRY" as they are not needed atm

File: README.md [hunk 6]
```diff
@@ -80,7 +79,7 @@ create a user and assign it rights:
 ```
 
 If you used the new `openshift-installer`, it creates a user named `kubeadmin`
-    with the password in the file `installer/auth/kubeadmin_password`.
+with the password in the file `installer/auth/kubeadmin_password`.
 
 To push the image to the remote registry, set `REMOTE_REGISTRY=true`, `PUSH_USER=kubeadmin`,
```
Comment (jcantrill) [inline README.md]: `make deploy-image` does this

File: Makefile (hunk 4 above)
Comment (alanconway) [reply Makefile]: +1 I only left it there because I assumed it was a CI requirement. Will remove.

File: README.md (hunk 6 above)
Comment (alanconway) [reply README.md]: +1 check updated README.

**Comment by alanconway** [comment]
Fixed all comments
/lgtm

**Comment by ewolinetz** [comment]
/hold
so it doesn't keep spinning

File: Makefile [hunk 7]
```diff
@@ -1,62 +1,38 @@
...
-APP_REPO=github.com/openshift/$(APP_NAME)
-TARGET=$(TARGET_DIR)/bin/$(APP_NAME)
-IMAGE_TAG?=quay.io/openshift/origin-$(APP_NAME):latest
-export IMAGE_TAG
-MAIN_PKG=cmd/manager/main.go
-export OCP_VERSION?=$(shell basename $(shell find manifests/  -maxdepth 1  -not -name manifests -type d))
-export CSV_FILE=$(CURDIR)/manifests/$(OCP_VERSION)/cluster-logging.v$(OCP_VERSION).0.clusterserviceversion.yaml
+export IMAGE_TAG?=quay.io/openshift/origin-$(APP_NAME):latest
+
+export OCP_VERSION?=4.5
```
Comment (jcantrill) [inline Makefile:11]: It needs a rebase but we are trying to move away from hard coding the version? Why is this here?
